Documented some stuff (#24)
apace100/origins-docs: docs/types/entity_action_types/drop_inventory.md

---
title: Drop Inventory (Entity Action Type)
date: 2022-6-07
---

# Drop Inventory

[Entity Action Type](../entity_action_types.md)

> Drops items directly from any spot in a player's inventory.

Reviewer: ```suggestion
Drops items from either the entity's inventory or a power that uses the [Inventory (Power Type)](../../power_types/inventory.md).
```
This entity action type is generalized to entities that can have inventories. It also works for powers that's using the `inventory` power type

Type ID: `origins:drop_inventory`

### Fields

Field  | Type | Default | Description
-------|------|---------|-------------
`inventory_type` | [Identifier](../../data_types/identifier.md) | `inventory` | Takes one of either `inventory` or `power`. Determines whether you're dropping items from the player inventory, or the extra inventory from the [`origins:inventory`](../types/power_types/inventory.md) power type respectively.

> `entity_action` | [Entity Action](../types/entity_action_types.md) | _optional_ | If specified, this action will be ran after the items are dropped.

Reviewer: ```suggestion
`entity_action` | [Entity Action](../types/entity_action_types.md) | _optional_ | If specified, this action will be executed on the entity **before** the items are dropped.
```
The entity action type is actually executed *before* the item action type and the item action type is executed *before* dropping the item

> `item_action` | [Item Action](../types/item_action_types.md) | _optional_ | If specified, this action will be ran on all affected items.

Reviewer: ```suggestion
`item_action` | [Item Action](../types/item_action_types.md) | _optional_ | If specified, this action will be executed on the affected items **before** they are dropped.
```

`item_condition` | [Item Condition](../types/item_condition_types.md) | _optional_ | If specified, only items which fulfill this condition will be dropped.
`slot` | [Identifier](../../data_types/identifier.md) | _optional_ | If specified, only items in the designated slot will be dropped. All valid inputs can be found [here](https://minecraft.fandom.com/wiki/Slot#Command_argument)
`slots` | [Array](../data_types/array.md) of [Identifiers](../data_types/identifier.md) | _optional_ | If specified, only items in the designated slots will be dropped. All valid inputs can be found [here](https://minecraft.fandom.com/wiki/Slot#Command_argument)

> `power` | [Identifier](../data_types/identifier.md) | _optional_ | Used to specify which power will be affected if the `inventory_type` field is set to `power`.

Reviewer: ```suggestion
`power` | [Identifier](../data_types/identifier.md) | _optional_ | If specified, this power will have its contents dropped as long as its using the `inventory` power type and the `inventory_type` field of this action is set to `"power"`.
```

`throw_randomly` | [Boolean](../data_types/boolean.md) | `false` | If true, items will be thrown in random directions instead of being normally dropped, similar to how items are dropped when you die.

> `retain_ownership` | [Boolean](../data_types/boolean.md) | `true` | If true, dropped items will have their `Owner` tag match the UUID of the player who the item originally belonged to, preventing others from picking it up before the original owner has a chance. 

Reviewer: ```suggestion
`retain_ownership` | [Boolean](../data_types/boolean.md) | `true` | If `true`, the dropped items will have their `Thrower` NBT set as the `UUID` NBT of the entity that invoked the action.
```
It actually only sets the `Thrower` NBT of the item entity, despite its name. The name of the fields are based on the actual name of the parameters from the [`PlayerEntity#dropItem`](https://maven.fabricmc.net/docs/yarn-1.19+build.1/net/minecraft/entity/player/PlayerEntity.html#dropItem(net.minecraft.item.ItemStack,boolean,boolean)) method
